File: Shared/Blog.Contracts/V1/Responses/CommentsResponses/PagedCommentsResponse.cs
namespace Blog.Contracts.V1.Responses.CommentsResponses;

using System.Collections.Generic;

/// <summary>
/// Paged comments response.
/// </summary>
public class PagedCommentsResponse
{
    /// <summary>
    /// Gets or sets the comments.
    /// </summary>
    /// <value>
    /// The comments.
    /// </value>
    public IList<CommentResponse> Comments { get; set; }

    /// <summary>
    /// Gets or sets the page information.
    /// </summary>
    /// <value>
    /// The page information.
    /// </value>
    public PageInfoResponse PageInfo { get; set; }
}

File: Shared/Blog.FSharp.Contracts/V1/Responses/CommentsResponses/PagedCommentsResponse.fs
namespace Blog.FSharp.Contracts.V1.Responses.CommentsResponses

open System.Collections.Generic
open Blog.FSharp.Contracts.V1.Responses

/// <summary>
/// Paged comments response.
/// </summary>
type PagedCommentsResponse() =
    /// <summary>
    /// Gets or sets the comments.
    /// </summary>
    member val Comments: IList<CommentResponse> = ResizeArray<CommentResponse>() :> IList<CommentResponse> with get, set

    /// <summary>
    /// Gets or sets the page information.
    /// </summary>
    member val PageInfo: PageInfoResponse = PageInfoResponse() with get, set

